src/Workspaces/Core/Portable/Workspace/Solution/LoadableTextAndVersionSource.cs

```diff
@@ -31,6 +46,67 @@ private Task<TextAndVersion> LoadAsync(CancellationToken cancellationToken)
 
         private TextAndVersion LoadSynchronously(CancellationToken cancellationToken)
             => Source.Loader.LoadTextSynchronously(Options, cancellationToken);
+
+        public bool TryGetValue([MaybeNullWhen(false)] out TextAndVersion value)
+        {
+            value = _instance;
+            if (value != null)
+                return true;
+
+            return _weakInstance != null && _weakInstance.TryGetTarget(out value) && value != null;
```

Comment:
Does `?.` work in this case?

```suggestion
            return _weakInstance?.TryGetTarget(out value) && value != null;
```

Reply:
i believe no *cry*.  you need an `== true`.  I personally have not found either approach to be 'better' in terms of preference.

Reply:
switched to the ?. form.